src/frontend/src/helpers/Storage/Interface.ts

```diff
@@ -8,34 +8,42 @@ const defaultSettings: Settings = {
   tab_width: 2
 };
 
-export {AbstractStorage,
+export {AbstractStorage, AbstractWebStorage,
         File, FileID, FileBrief,
         Project, ProjectID, ProjectBrief,
         Settings, defaultSettings}
 
 
 abstract class AbstractStorage {
   // projects
-  public abstract async newProject(name: string): Promise<void>;
+  public abstract async newProject(name: string): Promise<ProjectID>;
   public abstract async getProject(proj: ProjectID): Promise<Project>;
   public abstract async getProjects(): Promise<ProjectBrief[]>;
   public abstract async deleteProject(proj: ProjectID): Promise<void>;
-  public abstract async getFiles(proj: ProjectID): Promise<FileBrief[]>;
+  public abstract async getProjectFiles(proj: ProjectID): Promise<FileBrief[]>;
   // files
   public abstract async newFile(proj: ProjectID, filename: string, contents?: string): Promise<FileID>;
   public abstract async readFile(file: FileID): Promise<File>;
   public abstract async writeFile(file: FileID, contents: string): Promise<void>;
```

Comment:
writeFile should return the next FileID (as writing a file ==> new revision/file ID).  For now you can just return the current fileID.